### ISSUE TITLE
Do i still need to use Implicit wait?

### ISSUE THREAD

**Issue by abhinaba1080** [alfonso-presa/protractor-testability-plugin#10]
Hi,
here is a code sample:

```
await browser.findElement(by.css('123')).click();
await browser.refresh();
const c= await browser.findElement(by.css('345')).isPresent())
```

I am observing one behavior that, while browser is refreshing, isPresent() command in next line is not waiting for the page to be reloaded. It executes while page is still loading. 

I have set the testability plugin in my conf.ts, like:

```
 {
      package: 'protractor-testability-plugin'
  }
```

If i use implicit wait , issue is get resolved. But, whats the point of testability plugin if i need to set an implicit wait to the browser life-cycle?

Please guide me what is the better approach in this scenario.

**Comment by alfonso-presa** [comment]
Hi @abhinaba1080,

the testability plugin is based in detecting all asynchronous operations that happen in the browser while in the same page (i.e. Ajax calls, timeouts, etc...). It's specially meant for SPA applications and I'm sorry to say it will not help when page gets reloaded, as the listeners the plugin places in the page get removed when reloaded.

What I don't get is why you're requesting a browser.refresh. Wouldn't the click above perform the navigation? Normally when the browser is navigating between pages, or while a page is getting loaded, the browser will wait automatically for it to load (although it will not wait for async operations). After that load testability will get loaded automatically by protractor, even though there is a change of loosing some events there.

To sum up, if you're not testing a pure SPA application, that is, if you navigate or refresh pages, I would recommend using a more traditional approach and wait for some known element to be present for an amount of time. Tough you will be able to still use testability when operating in the same page.

**Comment by abhinaba1080** [comment]
Hi @alfonso-presa , right now we can set the protractor-testability-plugin as a plugin in protractor conf file. That means, it will only get loaded when web page loads for the first time. Lets say there are 3 pages, and I want to load protractor-testability-plugin after each page load, is it possible anyway?

**Comment by alfonso-presa** [comment]
@abhinaba1080 I think the "onPageLoad" event should raise on every page navigation and not only at the begining. Anyway, if it's not loading in your use case, as a workaround, I guess you should be able to perform a `require('protractor-testability-plugin').onPageLoad();` to force the plugin initialization for your page.

**Comment by abhinaba1080** [comment]
Hi this works fine for chrome.

However I am observing issues in IE11 and Firefox. Test execution became slow. I believe this plugin is not ment for IE11. Can you please confirm?

**Comment by alfonso-presa** [comment]
If you look at the automatic tests executed every week for this repository you'd see that they are executed against several browsers, one of them is ie11, and Firefox is other. If you are having any specific issues, please open an issue with the details.
